community/cloud-foundation/templates/dataproc/tests/integration/dataproc.bats

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bats
+
+source tests/helpers.bash
+
+TEST_NAME=$(basename "${BATS_TEST_FILENAME}" | cut -d '.' -f 1)
+
+# Create a random 10-char string and save it in a file.
+RANDOM_FILE="/tmp/${CLOUD_FOUNDATION_ORGANIZATION_ID}-${TEST_NAME}.txt"
+if [[ ! -e "${RANDOM_FILE}" ]]; then
+    RAND=$(head /dev/urandom | LC_ALL=C tr -dc a-z0-9 | head -c 10)
+    echo ${RAND} > "${RANDOM_FILE}"
+fi
+
+# Set variables based on the random string saved in the file.
+# envsubst requires all variables used in the example/config to be exported.
+if [[ -e "${RANDOM_FILE}" ]]; then
+    export RAND=$(cat "${RANDOM_FILE}")
+    DEPLOYMENT_NAME="${CLOUD_FOUNDATION_PROJECT_ID}-${TEST_NAME}-${RAND}"
+    # Replace underscores with dashes in the deployment name.
+    DEPLOYMENT_NAME=${DEPLOYMENT_NAME//_/-}
+    CONFIG=".${DEPLOYMENT_NAME}.yaml"
+    export CLUSTER_NAME="dataproc-cluster-${RAND}"
+    export ZONE="us-central1-a"
+    export TAG="test-tag-${RAND}"
+    export MASTER_INSTANCES="1"
+    export MASTER_TYPE="n1-standard-8"
+    export MASTER_DISK_SIZE="100"
+    export MASTER_DISK_TYPE="pd-ssd"
+    export WORKER_INSTANCES="2"
+    export WORKER_TYPE="n1-standard-4"
+    export SECONDARY_WORKER_INSTANCES="1"
+    export SECONDARY_WORKER_PREEMPTIBLE="true"
+fi
+
+########## HELPER FUNCTIONS ##########
+
+function create_config() {
+    echo "Creating ${CONFIG}"
+    envsubst < ${BATS_TEST_DIRNAME}/${TEST_NAME}.yaml > "${CONFIG}"
+}
+
+function delete_config() {
+    echo "Deleting ${CONFIG}"
+    rm -f "${CONFIG}"
+}
+
+function setup() {
+    # Global setup; executed once per test file.
+    if [ ${BATS_TEST_NUMBER} -eq 1 ]; then
+        create_config
+    fi
+
+    # Per-test setup steps.
+}
+
+function teardown() {
+    # Global teardown; executed once per test file.
+    if [[ "$BATS_TEST_NUMBER" -eq "${#BATS_TEST_NAMES[@]}" ]]; then
+        rm -f "${RANDOM_FILE}"
+        delete_config
+    fi
+
+    # Per-test teardown steps.
+}
+
+
```

Comment:
All tests are failing; I can't see where `status` is being set; did you mean `$?` ? 

```
 ✗ Creating deployment ocsi-courseralab1-guestbook-dataproc-4vqqszpdvh from .ocsi-courseralab1-guestbook-dataproc-4vqqszpdvh.yaml
   (in test file templates/dataproc/tests/integration/dataproc.bats, line 71)
     `[[ "$status" -eq 0 ]]' failed
   Creating .ocsi-courseralab1-guestbook-dataproc-4vqqszpdvh.yaml
 ✗ Verifying that dataproc-cluster-4vqqszpdvh settings are correct
   (in test file templates/dataproc/tests/integration/dataproc.bats, line 78)
     `[[ "$status" -eq 0 ]]' failed
 ✗ Verifying that master group settings are correct
   (in test file templates/dataproc/tests/integration/dataproc.bats, line 87)
     `[[ "$status" -eq 0 ]]' failed
 ✗ Verifying that worker group settings are correct
   (in test file templates/dataproc/tests/integration/dataproc.bats, line 98)
     `[[ "$status" -eq 0 ]]' failed
 ✗ Verifying that secondary worker group settings are correct
   (in test file templates/dataproc/tests/integration/dataproc.bats, line 109)
     `[[ "$status" -eq 0 ]]' failed
```

Reply:
`$status` and `$output` vars are introduced by `run` command. I'd say the test is failing either because Dataproc API is not enabled for your project, or because Deployment Manager's SA doesn't have `dataproc.worker` role. As far as I know, both are defaults.
I put that into the README, but there's a slight chance that I could've missed something extra. After all, it's hard to look for missing permissions while being organization admin :(

Reply:
I am sure dataproc API is enabled and I have that permission in my DM SA - I will rerun the test in a bit to be sure but the problem doesn't seem to be permission based and I am pretty sure the actual deployment created ok; I will recheck to be sure

Reply:
If it fails again, could you please add `echo "$output" > ~/output.txt` after `run` command of any failed test and rerun the suite? BATS doesn't output enough of the details about the failure (close to zero), and `echo` is the simplest way I know how to debug this stuff. `echo` destination has to be a file, otherwise BATS will swallow it.

Reply:
thanks for the hint! it turns out that my CPU quota was maxed out in the region we were running the tests. the output hint really helped! :+1: